.github/workflows/tests.yml

```diff
@@ -374,7 +374,7 @@ jobs:
           rc=0
           results=$(jq -r 'to_entries[] | [.key,.value.result] | join(" ")' <<< $NEEDS_CONTEXT)
           while read job result ; do
-              if [ "$result" != "success" ]; then
+              if [ "$result" != "success" ] || [ "$result" != "skipped" ]; then
```

Comment:
hrm, this looks risky.

If a job is cancelled, all the steps will be skipped. We don't really want to accept such test runs.

Reply:
Hmm, looks like some stuff returns cancelled, and others skipped? I guess that's the difference between those that were already running and those that hadn't? c.f. https://github.com/matrix-org/synapse/runs/3241552601

Reply:
yeah. I'm a bit worries about a race condition where the run is cancelled at just the right time so that the first set of jobs run to completion, but the second set all fail.

Could we only allow `lint-newsfile` to be skipped for now?

Reply:
> Could we only allow `lint-newsfile` to be skipped for now?

I have done this because all other options I can see are *awful*